src/java.desktop/share/classes/javax/swing/plaf/metal/MetalLookAndFeel.java

```diff
@@ -252,6 +252,9 @@ public boolean getSupportsWindowDecorations() {
      * used. For example, metal does not have a class named {@code
      * "MetalColorChooserUI"}, as such, {@code
      * javax.swing.plaf.basic.BasicColorChooserUI} is used.
+     * Also, class {@code MetalMenuBarUI} is used only for {@code OceanTheme}
```

Comment:
Maybe we can make it more generic, like some classes could be used only by some themes and registered only if that theme is "active", for example {@code MetalMenuBarUI} .... etc

Reply:
ok..we can do that...modified...